### PR TITLE
chore: add static file helpers on StorageWriter

### PR DESCRIPTION
**Pull request by Rjected**
This adds some helper methods to write headers / transactions to static files, similar to the `append_receipts_from_blocks` method, in `StorageWriter`. This makes the `log_transactions` method basically a wrapper over `StorageWriter` methods now

### CODE REVIEW

File: crates/storage/provider/src/writer/mod.rs [hunk 1]
```diff
@@ -138,11 +142,106 @@ impl<'a, 'b, DB: Database> StorageWriter<'a, 'b, DB> {
...
+                bodies_cursor.seek_exact(block_number)?.map(|(_, indices)| indices.first_tx_num());
+
+            // If there are no indices, that means there have been no transactions
+            //
+            // So instead of returning an error, use zero
+            if block_number == initial_block_number && first_tx_index.is_none() {
+                first_tx_index = Some(0);
+            }
+
+            // TODO: I guess this error will never be returned
```
Comment (mattsse) [inline crates/storage/provider/src/writer/mod.rs:218]: unclear what this todo means

File: crates/engine/tree/src/persistence.rs [hunk 2]
```diff
@@ -135,39 +135,31 @@ impl<DB: Database> PersistenceService<DB> {
...
     /// The [`update_transaction_meta`](Self::update_transaction_meta) method should be called
     /// after this, to update the checkpoints for headers and block bodies.
-    fn log_transactions(
-        &self,
-        block: Arc<SealedBlock>,
-        start_tx_number: u64,
-        td: U256,
-    ) -> ProviderResult<u64> {
-        debug!(target: "tree::persistence", ?td, ?start_tx_number, "Logging transactions");
+    fn log_transactions(&self, block: Arc<SealedBlock>) -> ProviderResult<u64> {
```
Comment (mattsse) [inline crates/engine/tree/src/persistence.rs:142]: I find `log_` very confusing can we stick to `write` ?

**Review by mattsse — CHANGES_REQUESTED** [review]
naming nits

File: crates/engine/tree/src/persistence.rs (hunk 2 above)
Comment (Rjected) [reply crates/engine/tree/src/persistence.rs]: sure, renamed

File: crates/storage/provider/src/writer/mod.rs (hunk 1 above)
Comment (Rjected) [reply crates/storage/provider/src/writer/mod.rs]: removed

**Review by mattsse — APPROVED** [review]

**Pull request merged**